### PR TITLE
Problem: dependency `github.com/enigmampc/btcutil` is vulnerable

### PR DESCRIPTION
**Pull request by ysong42**
Problem: 

This issue is first spotted in `chain-indexing`, https://github.com/crypto-com/chain-indexing/issues/661.

`github.com/enigmampc/btcutil` is vulnerable now and we need to remove it from our dependencies.

Solution:

Replace it with `github.com/scrtlabs/btcutil`.

👮🏻👮🏻👮🏻 !!!! REFERENCE THE PROBLEM YOUR ARE SOLVING IN THE PR TITLE AND DESCRIBE YOUR SOLUTION HERE !!!! DO NOT FORGET !!!! 👮🏻👮🏻👮🏻

# PR Checklist:

- [ ] Have you read the [CONTRIBUTING.md](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md)?
- [ ] Does your PR follow the [C4 patch requirements](https://rfc.zeromq.org/spec:42/C4/#23-patch-requirements)?
- [ ] Have you rebased your work on top of the latest master? 
- [ ] Have you checked your code compiles? (`make`)
- [ ] Have you included tests for any non-trivial functionality?
- [ ] Have you checked your code passes the unit tests? (`make test`)
- [ ] Have you checked your code formatting is correct? (`go fmt`)
- [ ] Have you checked your basic code style is fine? (`golangci-lint run`)
- [ ] If you added any dependencies, have you checked they do not contain any known vulnerabilities? (`go list -json -m all | nancy sleuth`)
- [ ] If your changes affect the client infrastructure, have you run the integration test?
- [ ] If your changes affect public APIs, does your PR follow the [C4 evolution of public contracts](https://rfc.zeromq.org/spec:42/C4/#26-evolution-of-public-contracts)?
- [ ] If your code changes public APIs, have you incremented the crate version numbers and documented your changes in the [CHANGELOG.md](https://github.com/crypto-org-chain/chain-main/blob/master/CHANGELOG.md)?
- [ ] If you are contributing for the first time, please read the agreement in [CONTRIBUTING.md](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md) now and add a comment to this pull request stating that your PR is in accordance with the [Developer's Certificate of Origin](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md#developer-certificate-of-originn).

Thank you for your code, it's appreciated! :)

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -156,6 +156,8 @@ replace (
 	// TODO: remove when middleware will be implemented
 	github.com/cosmos/ibc-go/v2 => github.com/crypto-org-chain/ibc-go/v2 v2.0.1-hooks
 
+	github.com/enigmampc/btcutil => github.com/scrtlabs/btcutil v1.0.3-0.20200723161021-e2fb6adb2a25
```
Comment (yihuang) [inline go.mod:159]: it use the same commit hash, how does it fix the vulnerability?

Comment (ysong42) [reply go.mod:159]: I think the issue is mainly on the package path.

The user/organization of `github.com/enigmampc` changed its name to `github.com/scrtlabs`. Therefore `github.com/enigmampc` could be taken by others.

The code itself is not vulnerable, just the package path.

Comment (ysong42) [reply go.mod:159]: image

The information I got from security team. And they asked to replace the `github.com/enigmampc/btcutil` with `github.com/scrtlabs/btcutil`.

Comment (Muggle-Du) [reply go.mod:159]: github will redirect github.com/enigmampc/btcutil to github.com/scrtlabs/btcutil, while github.com/enigmampc would not be redirected to github.com/scrtlabs, this means the organization enigmampc can be registered by hackers to inject some malicious codes

Comment (Muggle-Du) [reply go.mod:159]: an opposite example is github.com/lazyledger which is github.com/Celestia now

Comment (yihuang) [reply go.mod:159]: I see 👍

Comment (yihuang) [reply go.mod:159]: Is there an upstream PR to fix it?

Comment (ysong42) [reply go.mod:159]: Many modules in used will introduced this vulnerable dependency to `go.sum`.

If not doing the `replace` way, we may need to update a few dependencies: 
- `github.com/cosmos/ibc-go` to v2.0.2.
- `github.com/tharsis/ethermint` to v0.10.0-alpha1.
- `github.com/peggyjv/gravity-bridge/module`, we need to create a version on our own

Comment (yihuang) [reply go.mod:159]: > Is there an upstream PR to fix it in dependency?

it'll be fixed if we upgrade ethermint dependency, but we haven't decided yet if we use ibc-go v3 or v2 yet.

**Review by yihuang — APPROVED** [review]

**Pull request merged**